Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift

```diff
@@ -0,0 +1,379 @@
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+    
+    private lazy var _originalImage: UIImage? = {
+        switch config.type {
+        case .primerTestPayPal:
+            return UIImage(named: "paypal-logo-1", in: Bundle.primerResources, compatibleWith: nil)
+        case .primerTestSofort:
+            return UIImage(named: "sofort-logo", in: Bundle.primerResources, compatibleWith: nil)
+        default:
+            return buttonImage
+        }
+    }()
+    
+    lazy var submitButton: PrimerButton = {
+        let submitButton = PrimerButton()
+        submitButton.translatesAutoresizingMaskIntoConstraints = false
+        submitButton.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        submitButton.isAccessibilityElement = true
+        submitButton.accessibilityIdentifier = "submit_btn"
+        submitButton.isEnabled = false
+        submitButton.setTitleColor(theme.mainButton.text.color, for: .normal)
+        submitButton.backgroundColor = theme.mainButton.color(for: .disabled)
+        submitButton.layer.cornerRadius = 4
+        submitButton.clipsToBounds = true
+        submitButton.addTarget(self, action: #selector(payButtonTapped(_:)), for: .touchUpInside)
+        return submitButton
+    }()
+    
+    // MARK: - Deinit
+    
+    deinit {
+        log(logLevel: .debug, message: "🧨 deinit: \(self) \(Unmanaged.passUnretained(self).toOpaque())")
+    }
+    
+    // MARK: - Overrides
+    
+    override var originalImage: UIImage? {
+        get {
+            _originalImage
+        }
+        set {
+            _originalImage = newValue
+        }
+    }
+    
+    override func validate() throws {
+        guard let decodedClientToken = ClientTokenService.decodedClientToken, decodedClientToken.isValid else {
+            let err = PrimerError.invalidClientToken(userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"], diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            throw err
+        }
+    }
+    
+    override func start() {
+        
+        self.didStartTokenization = {
+            self.submitButton.startAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = false
+        }
+        
+        self.didFinishTokenization = { err in
+            self.submitButton.stopAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = true
+        }
+        
+        self.didStartPayment = {
+            self.submitButton.startAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = false
+        }
+        
+        self.didFinishPayment = { err in
+            self.submitButton.stopAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = true
+            self.willDismissPaymentMethodUI?()
```

Comment:
Should they be called at the same time? Both `present` and `push` provide a finish callback to call the `didDismiss`.

Reply:
I've updated it @EvansPie - let me know if now makes more sense 🙏 
<img width="1087" alt="Screen Shot 2022-06-06 at 14 42 18" src="https://user-images.githubusercontent.com/3584008/172162645-6f222a6c-9067-4975-a100-4234034edbeb.png">